### DISCUSSION TITLE
Hide modal window with select form when no options available

### DISCUSSION THREAD

**Discussion by mikrosmile** [filamentphp/filament#3353] [Ideas]
Hello to all members and developers of a such great framework. 
I’m facing one problem and didn’t find a great solution so far. 

I have an action which loads data from api and parse it to modal window with select form, when I click needed option it loads data. 

So, the issue is that I also have a case when no data is loaded from api and I need to close the select option form and push notification that no data is available. 

Now if I don’t have any options available (function that parse data returns null), I see empty select form. How can I dismiss modal window with select form when options are empty?

**Comment by danharrin** [comment]
Well it's important to remember that an action's modal won't open if there is no form, so technically you could do something like

```php
->form(function () {
    $options = get options from api;

    if (! count($options)) {
        return [];
    }

    return [Select::make('select_name')->required()->options($options)]
})
```

And then you can use:

```php
->action(function ($action, $data) {
    if (! $action->hasFormSchema()) {
        Notification::make()->title('No options')->warning();

        return;
    }

    // ...
})
```

**Reply by mikrosmile** [reply]
That's just what I needed. Thanks, I was too close to solve it myself :)
